src/main/java/org/synyx/urlaubsverwaltung/statistics/vacationoverview/web/ApplicationForLeaveVacationOverviewViewController.java

```diff
@@ -47,25 +79,217 @@ public String applicationForLeaveVacationOverview() {
+        final var departments = departmentService.getAllowedDepartmentsOfPerson(signedInUser);
+        model.addAttribute("departments", departments);
+
+        final var fallbackDepartment = departments.isEmpty() ? "" : departments.get(0).getName();
+        final var selectedDepartmentName = hasText(department) ? department : fallbackDepartment;
+        model.addAttribute("selectedDepartment", selectedDepartmentName);
 
-        prepareDepartments(signedInUser, model);
+        final var startDate = getStartDate(year, month);
+        final var endDate = getEndDate(year, month);
 
         model.addAttribute("currentYear", ZonedDateTime.now(UTC).getYear());
-        model.addAttribute("currentMonth", ZonedDateTime.now(UTC).getMonthValue());
+        model.addAttribute("selectedYear", startDate.getYear());
+
+        String selectedMonth;
+        if (month == null) {
+            selectedMonth = String.valueOf(startDate.getMonthValue());
+        } else if (hasText(month)) {
+            selectedMonth = month;
+        } else {
+            selectedMonth = "";
+        }
+        model.addAttribute("selectedMonth", selectedMonth);
+
+        final var overviewPersons = getOverviewPersonsForUser(signedInUser, departments, selectedDepartmentName);
+        final var sickNotes = sickNoteService.getAllActiveByYear(year == null ? LocalDate.now(UTC).getYear() : year);
+
+        final var vacationsByEmail = new HashMap<String, List<Application>>();
+        for (Person p : overviewPersons) {
+            var apps = applicationService.getApplicationsForACertainPeriodAndPerson(startDate, endDate, p);
+            vacationsByEmail.put(p.getEmail(), apps);
+        }
+
+        var monthsByNr = new HashMap<Integer, AbsenceOverviewMonthDto>();
+
+        // `date` is increased by one month at the end of this outer loop
+        // the outer loop builds the absence overview view model for every user desired month (specific one or all 12 🐵)
+        var date = startDate;
+        while (date.isBefore(endDate) || date.isEqual(endDate)) {
+
+            // `monthDate` is increased by one day at the end of this inner loop
+            // this inner loop builds the monthly absence items for every person of the given department
+            var monthDate = date;
+            while (monthDate.getMonthValue() == date.getMonthValue()) {
+
+                // since `monthDate` is increased by one day at the end of the loop we have to check
+                // if we have to create the month view dto in the current loop iteration.
+                if (!monthsByNr.containsKey(date.getMonthValue())) {
+                    var monthViewPersons = new ArrayList<AbsenceOverviewMonthPersonDto>(overviewPersons.size());
+                    for (Person person : overviewPersons) {
+                        var p = new AbsenceOverviewMonthPersonDto(
+                            person.getFirstName(), person.getLastName(), person.getEmail(), new ArrayList<>());
```

Comment:
List.of?

Reply:
kurz erklärt warum, List.of ist immutable und dto's sollten nicht verändert werden